src/github.com/matrix-org/dendrite/cmd/syncserver-integration-tests/main.go

```diff
@@ -60,6 +61,23 @@ var exe = test.KafkaExecutor{
 	OutputWriter: os.Stderr,
 }
 
+var (
+	lastRequestMutex sync.Mutex
+	lastRequestErr   error
+)
+
+func setLastRequestError(err error) {
+	lastRequestMutex.Lock()
```

Comment:
I don't appreciate its sanctimonious tone. Removing the mutexes would make the intention clearer rather than having to "read the rest of this document to understand the behavior of your program" imho.

Nevertheless, that document makes it clear that memory accesses can be reordered in a way that makes the mutex necessary to create a memory barrier. For that, thanks for the link :).